github/PullRequest.py

```diff
@@ -508,249 +439,202 @@ def create_issue_comment(self, body):
 
     def create_review(
         self,
-        commit=github.GithubObject.NotSet,
-        body=github.GithubObject.NotSet,
-        event=github.GithubObject.NotSet,
-        comments=github.GithubObject.NotSet,
-    ):
-        """
-        :calls: `POST /repos/{owner}/{repo}/pulls/{number}/reviews <https://docs.github.com/en/rest/reference/pulls#reviews>`_
-        :param commit: github.Commit.Commit
-        :param body: string
-        :param event: string
-        :param comments: list
-        :rtype: :class:`github.PullRequestReview.PullRequestReview`
-        """
-        assert commit is github.GithubObject.NotSet or isinstance(commit, github.Commit.Commit), commit
-        assert body is github.GithubObject.NotSet or isinstance(body, str), body
-        assert event is github.GithubObject.NotSet or isinstance(event, str), event
-        assert comments is github.GithubObject.NotSet or isinstance(comments, list), comments
-        post_parameters = dict()
-        if commit is not github.GithubObject.NotSet:
+        commit: Opt[github.Commit.Commit] = NotSet,
+        body: Opt[str] = NotSet,
+        event: Opt[str] = NotSet,
+        comments: Opt[list[ReviewComment]] = NotSet,
+    ) -> github.PullRequestReview.PullRequestReview:
+        """
+        :calls: `POST /repos/{owner}/{repo}/pulls/{number}/reviews <https://docs.github.com/en/free-pro-team@latest/rest/pulls/reviews?apiVersion=2022-11-28#create-a-review-for-a-pull-request>`_
+        """
+        assert is_optional(commit, github.Commit.Commit), commit
+        assert is_optional(body, str), body
+        assert is_optional(event, str), event
+        assert is_optional(comments, list), comments
```

Comment:
```suggestion
        assert is_optional_list(comments, ReviewComment), comments
```

Reply:
`ReviewComment` is a `TypedDict`, should we do this?

Reply:
Otherwise the method would allow any kind of list, e.g. list of ints.

Reply:
I mean, maybe, `assert is_optional_list(comments, dict), comments`

Reply:
That is better than nothing, would approve that one!